Simplenote/NotesListController.swift

```diff
@@ -0,0 +1,213 @@
+import Foundation
+import CoreData
+import SimplenoteFoundation
+import SimplenoteSearch
+
+
+// MARK: - NotesListController
+//
+class NotesListController: NSObject {
+
+    /// Core Data Kung Fu
+    ///
+    private let viewContext: NSManagedObjectContext
+    private lazy var notesController = ResultsController<Note>(viewContext: viewContext,
+                                                               matching: state.predicateForNotes(filter: filter),
+                                                               sortedBy: state.descriptorsForNotes(sortMode: sortMode))
+
+    /// FSM: Active State
+    ///
+    private(set) var state: NotesListState = .results {
+        didSet {
+            guard oldValue != state else {
+                return
+            }
+
+            refreshEverything()
+        }
+    }
+
+    /// Filter: Applied when we're not in Search Mode
+    ///
+    var filter: NotesListFilter = .everything {
+        didSet {
+            guard oldValue != filter else {
+                return
+            }
+
+            refreshPredicates()
+        }
+    }
+
+    /// Refreshes the Internal State, so that Search Results matching the specified Keyword will be filtered out.
+    /// - Important: It's up to the caller to invoke `performFetch`!!
+    ///
```

Comment:
100% on purpose, in order to avoid a duplicate `performFetch()`.

**Will be revisited** up next, I'm merging NoteListFilter + NoteListState, as per your suggestion.

Thank you for spotting this!!